firstvoices/backend/search/utils/hydration_utils.py

```diff
@@ -0,0 +1,224 @@
+import logging
+
+from backend.models import DictionaryEntry, Image, Song, Story
+from backend.models.media import Audio, Video
+from backend.search.utils.constants import (
+    ELASTICSEARCH_DICTIONARY_ENTRY_INDEX,
+    ELASTICSEARCH_MEDIA_INDEX,
+    ELASTICSEARCH_SONG_INDEX,
+    ELASTICSEARCH_STORY_INDEX,
+)
+from backend.search.utils.object_utils import get_object_by_id
+from backend.serializers.dictionary_serializers import DictionaryEntryMinimalSerializer
+from backend.serializers.media_serializers import (
+    AudioSerializer,
+    ImageSerializer,
+    VideoSerializer,
+)
+from backend.serializers.song_serializers import SongMinimalSerializer
+from backend.serializers.story_serializers import StoryMinimalSerializer
+from backend.views.utils import get_select_related_media_fields
+from firstvoices.settings import ELASTICSEARCH_LOGGER
+
+
+def separate_object_ids(search_results):
+    # Separate the mixed search results to fetch objects from db
+    search_results_dict = {
+        ELASTICSEARCH_DICTIONARY_ENTRY_INDEX: [],
+        ELASTICSEARCH_SONG_INDEX: [],
+        ELASTICSEARCH_STORY_INDEX: [],
+        ELASTICSEARCH_MEDIA_INDEX: {
+            "audio": [],
```

Comment:
I think either option would be okay here but it wouldn't hurt to define them as constants just in case we decide to refactor or change the names down the road.

Reply:
Constants sound good to me. I don't think they need to be an enum, and I think they can be specific to the search code.

Alternative idea, just to consider: would it be simpler to just put them in separate indices so we don't have to track the type at all? The other model types are all separate. Is there a benefit to merging the media models into one index?

Reply:
Yes, so I had the idea of using separate indices as well. My reasoning for putting them in one index was that the document structure is same for all media files with just the difference in `type`.

Separate indices:
- Upside: is that we can get some performance improvement, I'm not sure by how much, since my understanding is that ES only scans the inverted index during a `filter` operation and does not scan all the documents, so increasing the document count should not have a major impact. 
- Downside: Extra methods and overhead to manage 2 more indices. There will not be a lot of repeated code, which is good, since the document structure and the CUD operations remains the same for all 3 indices. But, I guess there will be more conditional clauses in CUD methods related to this where the the index name is checked instead of type.

One index:
- Upside: Less code, same document structure.
- Downside: If `filter` operation does affect the performance majorly, we can move to separate indices. I'm positive it may have some impact, since it is an additional operation after choosing the index, but after researching a bit, still not sure if it's a major time taker.